src/ops/cast.cc

```diff
@@ -14,6 +14,7 @@
  */
 
 #include "flexflow/ops/cast.h"
+#include "flexflow/model.h"
```

Comment:
Why is `model.h` included here? Is there something weaker (maybe `cast_params.h`?) that we can use here instead? `model.h` pulls in and contains a lot, so this could potentially trigger a lot of potentially unnecessary rebuilds

Reply:
I believe this is because of `get_or_create_node` (and in general, `FFModel`). I used [this diff](https://github.com/flexflow/FlexFlow/pull/273/files#diff-1a3382f4659ec99ef9690c3a4158c6a71732dc1b6641d03796bb797ed45fbd0c) for reference. `model.h` is removed from `cast.h` and instead included in `cast.cc`.

Reply:
It looks like the discussion on that PR also seems to acknowledge how large `model.h` is -- maybe we can separate the node creation stuff? Not sure how to approach this but if `model.h` is what is causing the bulk of the rebuild then separating params may not contribute too much unless we resolve that issue.

Reply:
Ah yeah, I think I mixed up the `.cc` and `.h` files and though I was looking at the other one. The main goal is to cause fewer kernel rebuilds as they are much slower, so as long as only the `.cc` files and not the `.h` files include `model.h`, then the impact on search time shouldn't be too bad. In the long run I think the direction to go in is to pull the kernel logic out of the operator class itself (see <https://github.com/lockshaw/FlexFlow/blob/feat/candle_split/include/flexflow/ops/linear_kernels.h> for an example) so that the operator interface can be refactored freely without ever requiring a kernel rebuild.

Reply:
Yes, that makes sense to me.

Reply:
Tracking in #303